File: 160425_baekjoon_hanoi/1.c
#include <stdio.h>
int Cnt;
void hanoi(int n, int a, int b)
{
	int temp;
	if(n==1)
	{
		printf("Move %d, move from Fall%d, to Fall%d\n", n, a, b);
	}
	else
	{
		temp=6-a-b;
		hanoi(n-1, a, temp);
		printf("Move %d, move from Fall%d, to Fall%d\n", n, a, b);
		hanoi(n-1, temp, b);
	}
        Cnt++;
}
int main()
{
	int n;
	printf("Input disk number:");
	scanf("%d", &n);
	hanoi(n, 1, 2);
        printf("Count : %d\n",Cnt);
	return 0;

}

File: 160425_baekjoon_hanoi/1.py
n = input()

print(pow(2, n) - 1)


def hanoi(num, a, b):
    if num == 1:
        print
        "%d %d" % (a, b)
        return
    else:
        temp = 6 - a - b
        hanoi(num - 1, a, temp)
        print
        "%d %d" % (a, b)
        hanoi(num - 1, temp, b)


hanoi(n, 1, 3)
